hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/BatchEntitySelectFetchInitializer.java

```diff
@@ -147,7 +148,13 @@ public void initializeInstance(RowProcessingState rowProcessingState) {
 		}
 
 		persistenceContext.getBatchFetchQueue().addBatchLoadableEntityKey( entityKey );
-		toBatchLoad.put( entityKey, parentAccess.getInitializedInstance() );
+		List<Object> objects = toBatchLoad.get( entityKey );
+		if ( objects == null ) {
+			objects = new ArrayList<>();
+		}
+		objects.add( parentAccess.getInitializedInstance() );
+
+		toBatchLoad.put( entityKey, objects );
```

Comment:
this statement could be moved inside the if(objects==null) block

Reply:
Thanks @frode-carlsen for the suggestion